server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -444,6 +444,10 @@ public Analyzer indexAnalyzer() {
         return this.indexAnalyzer;
     }
 
+    public boolean containsBrokenAnalysis(String field) {
+        return this.indexAnalyzer.containsBrokenAnalysis(field);
```

Comment:
It's exposed as a plain Analyzer though, and the impl itself is package-private (because it gets set on the IndexWriter immediately but needs to be configured subsequently, so it's a semi-transparent wrapper).  It is annoying though - really what needs to happen is that lucene needs to fix things so that term vectors can't contain broken offsets...

Reply:
I see thanks for explaining